File: leetcode/python/0026_remove_duplicates_from_sorted_array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # loop through nums using i (0, len(nums)) and j (1, len(nums)):
        # -> if nums[i + 1] != nums[j]: swap them, i++
        # -> if nums[i + 1] == nums[j]: continue
        # j++ regardless
        i = 0
        for j in range(1, len(nums)):
            if nums[i] != nums[j]:
                nums[i + 1], nums[j] = nums[j], nums[i + 1]
                i += 1
        return i + 1

File: leetcode/java/0026_remove_duplicates_from_sorted_array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 1) return 1;

        int i = 0, j = 1;

        while (j < nums.length) {
            if (nums[i] != nums[j]) {
                int tmp = nums[i + 1];
                nums[i + 1] = nums[j];
                nums[j] = tmp;
                i++;
            }
            j++;
        }

        return i + 1;
    }
}